src/gui/search/searchwidget.cpp

```diff
@@ -248,10 +248,18 @@ void SearchWidget::addTorrentToSession(const QString &source)
 
 void SearchWidget::on_pluginsButton_clicked()
 {
-    PluginSelectDlg *dlg = new PluginSelectDlg(m_searchEngine, this);
-    connect(dlg, &PluginSelectDlg::pluginsChanged, this, &SearchWidget::fillCatCombobox);
-    connect(dlg, &PluginSelectDlg::pluginsChanged, this, &SearchWidget::fillPluginComboBox);
-    connect(dlg, &PluginSelectDlg::pluginsChanged, this, &SearchWidget::selectActivePage);
+    new PluginSelectDlg(m_searchEngine, this);
+
+    const auto onPluginChanged = [this]() {
+        fillCatCombobox();
+        fillPluginComboBox();
+        selectActivePage();
+    };
+
+    connect(m_searchEngine, &SearchEngine::pluginInstalled, onPluginChanged);
```

Comment:
Shouldn't you pass `context` to these conections? Now your lambda can be called (via SearchEngine signal) when dialog deleted.